src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -320,6 +320,24 @@ struct FlowResult
 };
 /// @endcond
 
+/// @cond INTERNAL
+inline boost::optional<Quality>
+qualityUpperBound(ReadView const& v, Strand const& strand)
+{
+    Quality q{STAmount::uRateOne};
+    boost::optional<Quality> stepQ;
+    DebtDirection dir = DebtDirection::issues;
+    for (auto const& step : strand)
+    {
+        if (std::tie(stepQ, dir) = step->qualityUpperBound(v, dir); stepQ)
+            q = composed_quality(q, *stepQ);
+        else
+            return boost::none;
+    }
+    return q;
+};
+/// @endcond
+
```

Comment:
A quick comment to explain moving this function: I have an upcoming PR that requires `qualityUpperBound` be be declared earlier in this file, so this PR moved it to reduce conflicts with the upcoming PR.